lib/watcher.js

```diff
@@ -173,10 +175,68 @@ Watcher.prototype.updateExclusivity = function (file, hasExclusiveTests) {
 	}
 };
 
+Watcher.prototype.trackFailures = function (api) {
```

Comment:
> The current approach is relatively clean

It's clean enough, but `watcher.js` as a whole is getting pretty complex. I'm just looking for ways to separate what it does into logical chunks.